be/src/storage/olap_server.cpp

```diff
@@ -176,10 +177,95 @@ Status StorageEngine::start_bg_threads() {
         }
     }
 
+    if (!config::disable_storage_page_cache) {
+        _adjust_cache_thread = std::thread([this] { _adjust_pagecache_callback(nullptr); });
+        Thread::set_thread_name(_adjust_cache_thread, "adjust_cache");
+    }
+
     LOG(INFO) << "All backgroud threads of storage engine have started.";
     return Status::OK();
 }
 
+void evict_pagecache(StoragePageCache* cache, int64_t bytes_to_dec, std::atomic<bool>& stoped) {
+    if (bytes_to_dec > 0) {
+        int64_t bytes = bytes_to_dec;
+        while (bytes >= GCBYTES_ONE_STEP) {
+            // Evicting 1GB of data takes about 1 second, check if process have been canceled.
+            if (UNLIKELY(stoped)) {
+                return;
+            }
+            cache->adjust_capacity(-GCBYTES_ONE_STEP, kcacheMinSize);
+            bytes -= GCBYTES_ONE_STEP;
+        }
+        if (bytes > 0) {
+            cache->adjust_capacity(-bytes, kcacheMinSize);
+        }
+    }
+}
+
+void* StorageEngine::_adjust_pagecache_callback(void* arg_this) {
+#ifdef GOOGLE_PROFILER
+    ProfilerRegisterThread();
+#endif
+    GCHelper dec_advisor(config::pagecache_adjuct_period, config::auto_adjust_pagecache_interval_seconds,
```

Comment:
What's the difference between config::pagecache_adjuct_period and config::auto_adjust_pagecache_interval_seconds?